### DISCUSSION TITLE
How do I generate heatmap-dotplot presented in SCENIC+ paper

### DISCUSSION THREAD

**Discussion by kennethho04** [aertslab/scenicplus#497] [Q&A]
Hi all,

In the SCENIC+ paper, TF regulators of mesenchymal and epithelial cell types were selected using a heatmap-dotplot where the color represents TF expression and the dot size represent eRegulon specificity score (RSS) before diving into the perturbation simulations (shown below). 

<img width="995" alt="image" src="https://github.com/user-attachments/assets/ee8546a4-fb34-48f5-b652-410339c45851">

This heatmap-dotplot is easier to interpret than the one presented in the new tutorial. However, the code that generated that figure (provided in the paper) seems to be based on an older version of SCENIC+. How can I approach generating a similar heatmap in the newer SCENIC+ that utilizes mudata instead of scplus_object? How can I obtain TF expression from the scplus_mudata? Would it just be the scRNA_count modality of the mudata? Is there a way I can incorporate the rss info from regulon_specificity_scores() into my mudata for this heatmap?

Best,
Kenneth

**Comment by SeppeDeWinter** [comment]
Hi @kennethho04 

Have you tried running the command provided in the tutorial?

https://scenicplus.readthedocs.io/en/latest/human_cerebellum.html#Heatmap-dotplot

All the best,

Seppe

**Reply by RebuffetLucas** [reply]
Dear Seppe,
First, thank you very much for this nice tool and your active responses on the GitHub page.
Regarding the previous question: In the new version of the tutorial there are two kind of activator: +/+ and +/- while in the paper and the previous version of the tutorial there is only the (+) category.
1-	How do you interpret +/- relationship (TF positively correlates with TG but genomic regions are closed) ? Should it just be filtered out ?
2-	Also, some regulons appear in both forms, for example NR3C1_direct_+/+ and NR3C1_+/- . How can we interpret that? What should we do in these case? Should we remove the +/- form to keep only the +/+ in the heatmap?
3-	Finally, for the repressor part, if we want to keep only the repressors that can be interpreted in the most straight forward way (e.g the regulon where the TF closes the chromatin and makes the RNA go down)  what would you recommend regarding the repressor TF ? Should we keep only the -/- or the -/+ ?

Thanks again for this very nice tool!
Best,
Lucas

**Reply by SeppeDeWinter** [reply]
Hi @RebuffetLucas 

Please see the following table from: https://scenicplus.readthedocs.io/en/latest/api.html.

Does this answer your question?


TF-to-gene relationship | region-to-gene relationship | biological role
-- | -- | --
positive (+) | positive (+) | TF opens chromatin and activates gene expression
positive (+) | negative (-) | When the TF is expressed the target gene is also expressed but the regions linked to the gene are closed.
negative (-) | positive (+) | When the TF is expressed the target gene is not expressed. When the target gene is expressed, regions linked to this gene are open. TF could be a chromatin closing repressor.
negative (-) | negative (-) | When the TF is expressed the target gene is not expressed. When the target gene is expressed, regions linked to this gene are closed.

**Reply by RebuffetLucas** [reply]
Dear Seppe,
Thanks for answering. I already saw this table but still found it hard to interpret my results. I still don't understand if the (+) regulons in your SCENIC+ paper correspond to +/+ only or to +/+ and +/- regulons.
For the repressor part, based on this table, I would only keep the -/+ to have chromatin closing repressors TF right?
Best,
Lucas

**Reply by SeppeDeWinter** [reply]
Ah like this!

Yes the (+) ones in the paper correspond to +/+ only.
And yes, chromatin closing repressors would be represented by `-/+`. But please take those predictions with a grain of salt, predicting negative interactions is far more difficult compared to predicting positive ones.

Best,

Seppe

**Reply by RebuffetLucas** [reply]
OK, thank you very much Seppe.
Last question if I may: I'm not sure how to interpret the +/- biologically speaking? How could a TF be correlated with rna gene expression if the corresponding regions are closed?
Best,
Lucas
